[conradsoon] iP

conradsoon opened the pull request:
##### DeterministicParrot

> “HFHKK FMGFMNFBSMS. BM NSB BMGSMBNGMNBGFNBMGMN SG S D FSFSDFDSDSFFSDFSSFFSDDFSNMSFDMNSFBMNFSBMSDFMNBDSFBMN S FFS DFMS MNBDFS BMNSDF BMNFSMNFS ” – me when doing the ip every week

DeterministicParrot frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here]().
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java{.no-line-numbers}
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

conradsoon committed:
Renamed Duke to DeterministicParrot, added basic greeting and goodbye functionality.

conradsoon committed:
Added echo command functionality, added some indentations for responses.

conradsoon committed:
Added the ability to store whatever text entered by the user and display them back to the user when requested.

conradsoon committed:
Add the ability to mark tasks as done and the ability to change the status back to not done.

conradsoon committed:
Refactor into shell-style command handling.

conradsoon committed:
Add support for tracking three types of tasks: todos, events, deadlines.

conradsoon committed:
Added automated testing for MacOS.

conradsoon committed:
Added error checking.

conradsoon committed:
Added delete functionality.

conradsoon committed:
Added save and load functionality.

conradsoon committed:
Merge branch-Level-7 into master

conradsoon committed:
Implemented datetime support.

conradsoon committed:
Merge branch-Level-8 into master

conradsoon committed:
Ignored data directory.

conradsoon committed:
Refactored Task, Event, Deadline and Todo into their own classes

conradsoon committed:
Took out DeterministicParrotException into its own class

conradsoon committed:
Refactored into Parser, Storage, Tasklist and Ui.

conradsoon committed:
Merge A-MoreOOP into master

conradsoon committed:
Moved into deterministicparrot package.

conradsoon committed:
Merge branch-A-Packages into master

conradsoon committed:
Added gradle support to project.

conradsoon committed:
Merge add-gradle-support into master

conradsoon committed:
Add unit tests for Event.

conradsoon committed:
Add integration test for entire DeterministicParrot

conradsoon committed:
Merge branch-A-JUnit into master

conradsoon committed:
Add shadowJar

conradsoon committed:
Merge branch-A-Jar into master

conradsoon committed:
Added JavaDocs

conradsoon committed:
Add missing access specifiers to constructors.

conradsoon committed:
Add find command for finding task

conradsoon committed:
Merge branch-A-JavaDoc into master

conradsoon committed:
Merged codingstandard into main.

conradsoon committed:
Merge branch-level-9 into main

raydenlim commented on src/main/java/deterministicparrot/DeterministicParrot.java:
```diff
@@ -0,0 +1,267 @@
+     *
+     * @param args The arguments passed to the deadline command.
+     * @throws Exception If an error occurs while adding the task.
+     */
```
Very elegant way of parsing the task + description to deadline & event!😄👍

conradsoon replied:
thank you :)

raydenlim commented on src/main/java/deterministicparrot/DeterministicParrot.java:
```diff
@@ -0,0 +1,267 @@
+        }
+    }
+
+    private void findTask(String args[]) throws Exception {
```
Not sure if this method handles the case where there are no matching tasks. 
If it does, im sorry, im a coding noob :(
but wow, stringbuilder throws me back to CS2040S days 😨

conradsoon replied:
if there's no matching tasks i think it just doesn't print any of the tasks. will check again

raydenlim commented on src/main/java/deterministicparrot/Storage.java:
```diff
@@ -0,0 +1,69 @@
+ */
+public class Storage {
+
+    private static final String DATA_FILE_PATH = "./data/data.txt";
```
good job following the coding standards 😃

raydenlim commented on src/main/java/deterministicparrot/TaskList.java:
```diff
@@ -0,0 +1,196 @@
+     * @return A deserialized TaskList object.
+     * @throws Exception If there's an error during deserialization.
+     */
+    public static TaskList deserialize(String rawData) throws Exception {
```
Good job parsing :)
Btw, if im not wrong, you dont need to indent the cases. 
E.g
```java
switch (data[0]) {
case "T":
    ToDo todo = new ToDo(data[2]);
    if (data[1].equals("1")) todo.markAsDone();
    tasks.add(todo);
    break;
case "D":
    Deadline deadline = new Deadline(data[2], data[3]);
    if (data[1].equals("1")) deadline.markAsDone();
    tasks.add(deadline);
    break;
case "E":
    String[] time = data[3].split(" ");
    Event event = new Event(data[2], time[0], time[1]);
    if (data[1].equals("1")) event.markAsDone();
    tasks.add(event);
    break;
}
```

conradsoon replied:
ah okay, thanks for that. will change later.

raydenlim commented on src/main/java/deterministicparrot/Ui.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Prints a dashed line separator.
+     */
+    public void printDash() {
```
maybe change to printLine() ? haha, cus 
this is a dash ----
this is an underline _______

**trying to hit the 4-5 comments for PR 😄 

Btw, I really enjoyed looking through your code, I learnt alot !! 🥂

conradsoon replied:
🥂🥂

conradsoon committed:
Add CheckStyle to project

conradsoon committed:
Merge branch-A-CheckStyle into master

conradsoon committed:
Add JavaFX to project

conradsoon committed:
Add GUI to project

conradsoon committed:
Add FXML to generate GUI for project

conradsoon committed:
Add logic to get text output from DeterministicParrot to GUI

XihuaZ commented on src/main/java/deterministicparrot/DPUtils.java:
```diff
@@ -0,0 +1,43 @@
+/**
+ * Utility class for date and time formatting and parsing in the Deterministic Parrot application.
+ */
+public class DPUtils {
```
Is it correct to have multiple consecutive capital letters? 

"Abbreviations and acronyms should not be uppercase when used as a part of a name." -- coding standard

conradsoon replied:
thank you so much! will fix accordingly.

XihuaZ reviewed (commented):
Coding standard is adhered to overall, however, do be careful that abbreviations and acronyms should not be uppercase when used as a part of a name.
Good job!

conradsoon committed:
Modify build.gradle and add image resources

The current project's build configuration lacks some necessary dependencies and resources.

To enhance the application's functionality and improve its UI, it's essential to incorporate new dependencies and assets.

Let's update the build.gradle file to include these dependencies and also introduce new image resources to the asset folder.

Adding these will improve app performance and provide a better user experience. The image resources include icons and background images essential for the app's UI/UX.

conradsoon committed:
Introduce assertions for robustness

Throughout the codebase, certain conditions assumed to be true can cause
unexpected behaviors if not met.

Incorporating assertions helps us enforce these conditions and brings
issues to the forefront during the development phase, preventing potential
runtime anomalies.

Let's,
* Add assertions in critical areas where assumptions are made
* Document the conditions we are asserting for clarity

Utilizing assertions will help in ensuring the software's robustness
and increase overall code quality.

conradsoon committed:
Merge pull request #2 from conradsoon/branch-A-Assertions

Introduce assertions for robustness

conradsoon committed:
Improve code quality by formatting according to CheckStyle

conradsoon committed:
Merge pull request #3 from conradsoon/branch-A-CodeQuality

Improve code quality by formatting according to CheckStyle

conradsoon committed:
Implement "upcoming" command, which allows you to list all deadlines before a certain time

conradsoon committed:
Merge pull request #4 from conradsoon/branch-B-Reminders

Implement "upcoming" command, which allows you to list all deadlines …

conradsoon committed:
Add Ui.png for docs

conradsoon committed:
Fix bug where tasks not being loaded in correctly

conradsoon committed:
Add readme for docs
